File: src/main/java/cz/vutbr/fit/interlockSim/context/ContextCreationException.java
/* Brno University of Technology
 * Faculty of Information Technology
 *
 * BSc Thesis  2006/2007
 *
 * Railway Interlocking Simulator
 *
 * Bedrich Hovorka
 */
package cz.vutbr.fit.interlockSim.context;

/**
 * Is throwing by {@link ContextFactory}, when source for context is wrong
 *
 */
public class ContextCreationException extends Exception {
	/**
	 * @see Exception#Exception()
	 *
	 */
	public ContextCreationException() {
		super();
	}

	/**
	 * @see Exception#Exception(Throwable)
	 * @param cause
	 */
	public ContextCreationException(Throwable cause) {
		super(cause);
	}

	/**
	 * @see Exception#Exception(String)
	 * @param string
	 */
	public ContextCreationException(String string) {
		super(string);
	}

}

File: core/src/commonMain/kotlin/cz/vutbr/fit/interlockSim/context/ContextCreationException.kt
/* Brno University of Technology
 * Faculty of Information Technology
 *
 * BSc Thesis  2006/2007
 *
 * Railway Interlocking Simulator
 *
 * Bedrich Hovorka
 */
package cz.vutbr.fit.interlockSim.context

/**
 * Is throwing by [ContextFactory], when source for context is wrong
 *
 */
class ContextCreationException : Exception {
	/**
	 * @see Exception.Exception
	 *
	 */
	constructor() : super()

	/**
	 * @see Exception.Exception
	 * @param cause
	 */
	constructor(cause: Throwable) : super(cause)

	/**
	 * @see Exception.Exception
	 * @param string
	 */
	constructor(string: String) : super(string)

	/**
	 * @see Exception.Exception
	 * @param message
	 * @param cause
	 */
	constructor(message: String, cause: Throwable?) : super(message, cause)
}
